_stbt/grid.py

```diff
@@ -272,3 +230,52 @@ def __next__(self):
 
     def next(self):  # Python 2
         return self.__next__()
+
+
+def grid_to_navigation_graph(grid):
+    """Generate a Graph that describes navigation between cells in the grid.
+
+    Creates a `networkx.DiGraph` (`Directed Graph`_) that models cells in the
+    grid as nodes in the graph. Each edge in the graph has a ``key`` attribute
+    set to "KEY_LEFT", "KEY_RIGHT", "KEY_UP", or "KEY_DOWN", corresponding to
+    the keypress that will move a selection from one node to another.
+
+    :param Grid grid: The grid to model. If the Grid has data associated
+        with the cells, each node in the graph will be named with the
+        corresponding cell's ``data``; otherwise the nodes are numbered
+        according to the cell's ``index``. This means that the cell's ``data``
+        must be `hashable`_ so that it can be used as a `networkx` node.
+
+    :returns: A `networkx.DiGraph`.
+
+    For example, to create a graph suitable as the ``graph`` parameter of
+    `stbt.Keyboard`::
+
+        grid = stbt.Grid(region, data=["ABCDEFG",
+                                       "HIJKLMN",
+                                       "OPQRSTU",
+                                       "VWXYZ-'"])
+        keyboard = stbt.Keyboard(grid_to_navigation_graph(grid))
+
+    .. _Directed Graph: https://en.wikipedia.org/wiki/Directed_graph
+    .. _hashable: https://docs.python.org/3.6/glossary.html#term-hashable
+    """
+    G = nx.DiGraph()
+
+    def name(c):
+        if c.data is None:
+            return c.index
+        else:
+            return c.data
```

Comment:
I'm not too fussed and I want to get this merged.

Grids with `region=None` are only really for our self-tests; in practice when you create a Grid you need to specify a region so that you can map the selection (found with `stbt.match`) to the relevant grid data.